src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1498,11 +1571,56 @@ public void Dispose()
             // Do this here because otherwise if a purge job runs there will
             // be a deadlock while the purge job waits for the lock that is held
             // by the disposal which is waiting for timer callbacks to finish
-            StopExpirePurgeTimer(); 
-            ThreadSafety.DoLocked(() =>
-            {
-                ThrowIfActiveItems();
-                Dispose(true);
+            StopExpirePurgeTimer();
+            ThreadSafety.DoLocked(() => {
+                if (IsClosed) {
+                    return;
+                }
+
+                if (!_isClosing) {
+                    _isClosing = true;
+                }
+            });
+
+            _closeCondition.Reset();
+            foreach (var q in ActiveLiveQueries) {
```

Comment:
What about stopping replicators?

Reply:
In the replicator dev PR as well (https://github.com/couchbase/couchbase-lite-net/pull/1205/files#diff-795d11b3c81c750fb0cd00714df85829R1720)